packages/components/src/table/index.js

```diff
@@ -91,6 +91,10 @@ class TableCard extends Component {
 		}
 	}
 
+	getShowCols( headers ) {
+		return headers.map( ( { key, hiddenByDefault } ) => ! hiddenByDefault && key ).filter( Boolean );
```

Comment:
When I was checking this out yesterday, I was a bit confused before remembering that we also use `hiddenByDefault` to determine hidden status and not just the default.

Completely optional, but thoughts on changing this to just `hidden`?

Reply:
I'm not 100% sure about this. In theory, a 3rd party could use the `<Table>` component without an `onColumnsChange()` prop. So columns' visibility is stored completely in the state but the `headers` prop is never updated. I think that's why that prop is named `hiddenByDefault`, because not necessarily it goes in sync with `state.showCols`.

Do you think that makes sense or is it too confusing?